app/mailers/host_report_mailer.rb

```diff
@@ -0,0 +1,52 @@
+require 'uri'
+
+class HostReportMailer < ApplicationMailer
+  helper ApplicationHelper
+
+  # sends out a summary email of hosts and their metrics (e.g. how many changes failures etc).
+  def summary(options = {})
+    raise ::Foreman::Exception, N_("Must specify a valid user with email enabled") unless (user = User.find(options[:user]))
+    hosts = User.as user do
+      Host::Managed.authorized_as(user, :view_hosts, Host).order(:last_report).limit(100)
+    end
+
+    @format = options[:format]
+    time = options[:time] || 1.day.ago
+    host_data = HostReport.summarise(time, hosts, @format)
+    total_metrics = load_metrics(host_data)
+    total = 0
+    total_metrics.each_value { |v| total += v }
+
+    @hosts = host_data
```

Comment:
Can you directly store into this attribute and pass this into `load_metrics`? No need for intermediate variable.